### PR TITLE
Improve docs for the `const` keyword

### PR DESCRIPTION
**Pull request by camelid**
@rustbot modify labels: T-doc

### CODE REVIEW

File: library/std/src/keyword_docs.rs [hunk 1]
```diff
@@ -128,27 +130,27 @@ mod break_keyword {}
...
 /// to which one should be used at which times. To put it simply, constants are inlined wherever
-/// they're used, making using them identical to simply replacing the name of the const with its
-/// value. Static variables on the other hand point to a single location in memory, which all
+/// they're used, making using them identical to simply replacing the name of the `const` with its
+/// value. Static variables, on the other hand, point to a single location in memory, which all
 /// accesses share. This means that, unlike with constants, they can't have destructors, and act as
 /// a single value across the entire codebase.
 ///
-/// Constants, as with statics, should always be in SCREAMING_SNAKE_CASE.
+/// Constants, as with statics, should always be in `SCREAMING_SNAKE_CASE`.
```
Comment (camelid) [inline library/std/src/keyword_docs.rs:146]: What do you think of this?

```suggestion
/// Constants, like statics, should always be in `SCREAMING_SNAKE_CASE`.
```

File: library/std/src/keyword_docs.rs [hunk 2]
```diff
@@ -107,18 +107,20 @@ mod break_keyword {}
...
-/// Constants must be explicitly typed, unlike with `let` you can't ignore its type and let the
-/// compiler figure it out. Any constant value can be defined in a const, which in practice happens
-/// to be most things that would be reasonable to have a constant (barring `const fn`s). For
-/// example, you can't have a File as a `const`.
+/// Constants must be explicitly typed; unlike with `let`, you can't ignore its type and let the
+/// compiler figure it out. Any constant value can be defined in a `const`, which in practice happens
+/// to be most things that would be reasonable to have in a constant (barring `const fn`s). For
+/// example, you can't have a [`File`] as a `const`.
+///
+/// [`File`]: crate::fs::File
```
Comment (camelid) [inline library/std/src/keyword_docs.rs:123]: Yay, intra-doc links! :)

File: library/std/src/keyword_docs.rs [hunk 3]
```diff
@@ -107,18 +107,20 @@ mod break_keyword {}
...
 /// const THING: u32 = 0xABAD1DEA;
 ///
 /// let foo = 123 + THING;
 /// ```
 ///
-/// Constants must be explicitly typed, unlike with `let` you can't ignore its type and let the
-/// compiler figure it out. Any constant value can be defined in a const, which in practice happens
-/// to be most things that would be reasonable to have a constant (barring `const fn`s). For
-/// example, you can't have a File as a `const`.
+/// Constants must be explicitly typed; unlike with `let`, you can't ignore its type and let the
```
Comment (jyn514) [inline library/std/src/keyword_docs.rs:118]:
```suggestion
/// Constants must be explicitly typed; unlike with `let`, you can't ignore their type and let the
```

File: library/std/src/keyword_docs.rs [hunk 4]
```diff
@@ -128,27 +130,27 @@ mod break_keyword {}
 /// const WORDS: &'static str = "hello rust!";
 /// ```
 ///
-/// Thanks to static lifetime elision, you usually don't have to explicitly use 'static:
+/// Thanks to static lifetime elision, you usually don't have to explicitly use `'static`:
```
Comment (jyn514) [inline library/std/src/keyword_docs.rs:133]:
```suggestion
/// Thanks to static lifetime elision, you usually don't have to explicitly use `static`:
```

File: library/std/src/keyword_docs.rs [hunk 5]
```diff
@@ -128,27 +130,27 @@ mod break_keyword {}
...
+/// value. Static variables, on the other hand, point to a single location in memory, which all
 /// accesses share. This means that, unlike with constants, they can't have destructors, and act as
 /// a single value across the entire codebase.
 ///
-/// Constants, as with statics, should always be in SCREAMING_SNAKE_CASE.
+/// Constants, as with statics, should always be in `SCREAMING_SNAKE_CASE`.
 ///
 /// The `const` keyword is also used in raw pointers in combination with `mut`, as seen in `*const
-/// T` and `*mut T`. More about that can be read at the [pointer] primitive part of the Rust docs.
+/// T` and `*mut T`. More about that usage can be read at the Rust docs for the [pointer primitive].
```
Comment (jyn514) [inline library/std/src/keyword_docs.rs:149]: Maybe something like this?
```suggestion
/// T` and `*mut T`. The documentation for [pointer primitives] has more information about raw pointers.
```

File: library/std/src/keyword_docs.rs (hunk 4 above)
Comment (camelid) [reply library/std/src/keyword_docs.rs:133]: Hmm, I feel like it's more clear the way it currently is. Why do you suggest removing the lifetime marker `'`?

File: library/std/src/keyword_docs.rs (hunk 5 above)
Comment (camelid) [reply library/std/src/keyword_docs.rs:149]: What about:

```suggestion
/// T` and `*mut T`. More about the usage of `const` in raw pointers can be read at the Rust docs for the [pointer primitive].
```

File: library/std/src/keyword_docs.rs (hunk 4 above)
Comment (jyn514) [reply library/std/src/keyword_docs.rs:133]: Oops, this is what happens when you do drive-by PRs 😆 I missed that it's a lifetime. It's fine as-is then.

File: library/std/src/keyword_docs.rs (hunk 5 above)
Comment (jyn514) [reply library/std/src/keyword_docs.rs:149]: Better than mine, but 'the usage of' feels very heavy-handed to me. Maybe just remove it altogether?
```suggestion
/// T` and `*mut T`. More about `const` in raw pointers can be read at the Rust docs for the [pointer primitive].
```

Comment (camelid) [reply library/std/src/keyword_docs.rs:149]: I feel like it's a bit clearer though. Maybe this?

```suggestion
/// T` and `*mut T`. More about `const` as used in raw pointers can be read at the Rust docs for the [pointer primitive].
```

But it's not a big deal either way.

Comment (jyn514) [reply library/std/src/keyword_docs.rs:149]: I like that last one, let's do that :) I've done enough nitpicking :laughing:

**Pull request merged**